### PR TITLE
gnome3.updateScript: Add freeze functionality

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
On stable releases, we will want to change the freeze parameter in `pkgs/desktops/gnome-3/update.nix` to true to limit the gnome update script to only bump patch versions.

###### Things done
- [x] Tested on `release-20.03` branch with `freeze = true`
- [x] Tested on master branch

cc @worldofpeace @hedning

### CODE REVIEW

**Review by worldofpeace — DISMISSED** [review]
Looks great, could you run `find-latest-version.py` through `black`?

File: pkgs/desktops/gnome-3/update.nix [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
-{ packageName, attrPath ? packageName, versionPolicy ? "odd-unstable" }:
+{ stdenv, pkgs, lib, writeScript, python3, common-updater-scripts }:
+{ packageName, attrPath ? packageName, versionPolicy ? "odd-unstable", freeze ? true }:
 
 let
   python = python3.withPackages (p: [ p.requests ]);
+  upperBoundFlag =
+    let
+      package = lib.getAttrFromPath (lib.splitString "." attrPath) pkgs;
+      minorVersion = lib.versions.minor package.version;
```
Comment (worldofpeace) [inline pkgs/desktops/gnome-3/update.nix:9]: If I ran update.nix just on `gnome3` attribute I got
```
error: attribute 'version' missing
```

I guess it's possible some packages may not have a `version` attribute.
Maybe use `builtins.parseDrvName` to get `version` from `name`?

Comment (worldofpeace) [reply pkgs/desktops/gnome-3/update.nix:9]:
```suggestion
      minorVersion = lib.versions.minor (builtins.parseDrvName package.name).version;
```

**Comment by worldofpeace** [comment]
Cool, so now we get an `error: list index 1 is out of bounds` out of bounds error at `lib.versions.minor`

**Comment by jtojnar** [comment]
libgudev?

**Comment by worldofpeace** [comment]
> libgudev?

Yes, that attribute gives out of bounds because its version `233`

**Review by worldofpeace — APPROVED** [review]
I ran update.nix on `gnome3` attribute on `master` without frozen without a problem :heavy_check_mark: 
I ran update.nix on `gnome3` attribute on `release-20.03` without a problem,
and packages were updated in the proper bound :heavy_check_mark:

**Pull request merged**